--- a/mpc/configure
+++ b/mpc/configure
@@ -10847,31 +10847,8 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for MPFR" >&5
 $as_echo_n "checking for MPFR... " >&6; }
 LIBS="-lmpfr $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include "mpfr.h"
-int
-main ()
-{
-mpfr_t x;  mpfr_init(x) ; mpfr_clear(x);
-
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-else
-
-        { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-        as_fn_error "libmpfr not found or uses a different ABI." "$LINENO" 5
-
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes (DARWIN_BOOTSTRAP_ASSUME_MPFR)" >&5
+$as_echo "yes (DARWIN_BOOTSTRAP_ASSUME_MPFR)" >&6; }
 # Check for a recent GMP
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for recent GMP" >&5
 $as_echo_n "checking for recent GMP... " >&6; }
