diff -ru a/gcc/c/Make-lang.in b/gcc/c/Make-lang.in
--- a/gcc/c/Make-lang.in
+++ b/gcc/c/Make-lang.in
@@ -142,7 +142,7 @@
 
 # If C is enabled, require the selftests to be run for it
 # at each stage of the build:
-selftest-c: s-selftest-c
+selftest-c:
 
 C_SELFTEST_FLAGS = -xc $(SELFTEST_FLAGS)
 C_SELFTEST_DEPS = cc1$(exeext) $(SELFTEST_DEPS)
diff -ru a/gcc/config/darwin.h b/gcc/config/darwin.h
--- a/gcc/config/darwin.h
+++ b/gcc/config/darwin.h
@@ -329,9 +329,7 @@
 # define DARWIN_RPATH_LINK ""
 # define DARWIN_SHARED_LIBGCC \
 "%:version-compare(!> 10.11 mmacosx-version-min= -lgcc_s.1.1)"
-# define DARWIN_SHARED_WEAK_ADDS \
-"%{%:version-compare(>= 10.11 mmacosx-version-min= -lemutls_w): \
- " DARWIN_HEAP_T_LIB "}"
+# define DARWIN_SHARED_WEAK_ADDS " "
 #endif
 
 
@@ -567,12 +565,7 @@
     %:version-compare(>< 10.5 10.6 mmacosx-version-min= -lgcc_s.10.5)	  \
   } -lgcc "
 
-#define DARWIN_WEAK_CRTS \
-"%{static-libgcc|static:						  \
-    %{%:version-compare(>= 10.6 mmacosx-version-min= -lemutls_w):	  \
-      " DARWIN_HEAP_T_LIB "} ;						  \
-   : -lemutls_w	" DARWIN_HEAP_T_LIB "					  \
-  }"
+#define DARWIN_WEAK_CRTS ""
 
 /* We specify crt0.o as -lcrt0.o so that ld will search the library path.  */
 
diff -ru a/gcc/configure b/gcc/configure
--- a/gcc/configure
+++ b/gcc/configure
@@ -12345,7 +12345,7 @@
 
 else
 
-$as_echo "#define rlim_t long" >>confdefs.h
+:
 
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
diff -ru a/gcc/cp/Make-lang.in b/gcc/cp/Make-lang.in
--- a/gcc/cp/Make-lang.in
+++ b/gcc/cp/Make-lang.in
@@ -246,7 +246,7 @@
 
 # If C++ is enabled, require the selftests to be run for it
 # at each stage of the build:
-selftest-c++: s-selftest-c++
+selftest-c++:
 
 CPP_SELFTEST_DEPS = cc1plus$(exeext) $(SELFTEST_DEPS)
 CPP_SELFTEST_FLAGS = -xc++ $(SELFTEST_FLAGS)
diff -ru a/gmp/Makefile.in b/gmp/Makefile.in
--- a/gmp/Makefile.in
+++ b/gmp/Makefile.in
@@ -573,7 +573,7 @@
 LIBGMPXX_LT_CURRENT = 11
 LIBGMPXX_LT_REVISION = 0
 LIBGMPXX_LT_AGE = 7
-SUBDIRS = tests mpn mpz mpq mpf printf scanf rand cxx demos tune doc
+SUBDIRS = mpn mpz mpq mpf printf scanf rand cxx
 
 # Put asl.h here for now.
 
diff -ru a/libcody/configure b/libcody/configure
--- a/libcody/configure
+++ b/libcody/configure
@@ -2501,7 +2501,7 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-#if __cplusplus != 201103
+#if __cplusplus < 201103
 #error "C++11 is required"
 #endif
 
@@ -2522,7 +2522,7 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-#if __cplusplus != 201103
+#if __cplusplus < 201103
 #error "C++11 is required"
 #endif
 
@@ -2542,7 +2542,7 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-#if __cplusplus > 201103
+#if __cplusplus < 201103
 #error "C++11 is required"
 #endif
 
diff -ru a/libgcc/config.host b/libgcc/config.host
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -280,7 +280,7 @@
   if test "x$enable_darwin_at_rpath" = "xyes"; then
     tmake_file="$tmake_file t-darwin-rpath "
   fi
-  extra_parts="crt3.o libd10-uwfef.a crttms.o crttme.o libemutls_w.a"
+  extra_parts="libd10-uwfef.a"
   ;;
 *-*-dragonfly*)
   tmake_file="$tmake_file t-crtstuff-pic t-libgcc-pic t-eh-dw2-dip"
@@ -740,14 +740,14 @@
 	tmake_file="$tmake_file i386/t-crtpc t-crtfm i386/t-msabi"
 	tm_file="$tm_file i386/darwin-lib.h"
 	extra_parts="$extra_parts crtprec32.o crtprec64.o crtprec80.o"
-	extra_parts="$extra_parts crtfastmath.o libheapt_w.a"
+	extra_parts="$extra_parts crtfastmath.o"
 	tmake_file="${tmake_file} i386/t-heap-trampoline"
 	;;
 x86_64-*-darwin*)
 	tmake_file="$tmake_file i386/t-crtpc t-crtfm i386/t-msabi"
 	tm_file="$tm_file i386/darwin-lib.h"
 	extra_parts="$extra_parts crtprec32.o crtprec64.o crtprec80.o"
-	extra_parts="$extra_parts crtfastmath.o libheapt_w.a"
+	extra_parts="$extra_parts crtfastmath.o"
 	tmake_file="${tmake_file} i386/t-heap-trampoline"
 	;;
 i[34567]86-*-elfiamcu)
diff -ru a/libgcc/configure b/libgcc/configure
--- a/libgcc/configure
+++ b/libgcc/configure
@@ -5075,16 +5075,7 @@
 
 # Check 32bit or 64bit.  In the case of MIPS, this really determines the
 # word size rather than the address size.
-cat > conftest.c <<EOF
-#if defined(__x86_64__) || (!defined(__i386__) && defined(__LP64__)) \
-    || defined(__mips64) || defined(__loongarch64)
 host_address=64
-#else
-host_address=32
-#endif
-EOF
-eval `${CC-cc} -E conftest.c | grep host_address=`
-rm -f conftest.c
 
 case ${host} in
 mips*-*-*)
@@ -5126,17 +5117,8 @@
 fi
 else
 
-if ac_fn_c_compute_int "$LINENO" "__GLIBC__" "glibc_version_major"        "#include <features.h>"; then :
-
-else
-  glibc_version_major=0
-fi
-
-if ac_fn_c_compute_int "$LINENO" "__GLIBC_MINOR__" "glibc_version_minor"        "#include <features.h>"; then :
-
-else
-  glibc_version_minor=0
-fi
+glibc_version_major=0
+glibc_version_minor=0
 
 fi
 
diff -ru a/libiberty/fopen_unlocked.c b/libiberty/fopen_unlocked.c
--- a/libiberty/fopen_unlocked.c
+++ b/libiberty/fopen_unlocked.c
@@ -72,7 +72,7 @@
 #include "config.h"
 #endif
 #include <stdio.h>
-#ifdef HAVE_STDIO_EXT_H
+#if defined(HAVE_STDIO_EXT_H) && !defined(__APPLE__)
 #include <stdio_ext.h>
 #endif
 
diff -ru a/libiberty/hashtab.c b/libiberty/hashtab.c
--- a/libiberty/hashtab.c
+++ b/libiberty/hashtab.c
@@ -44,8 +44,12 @@
 #include <string.h>
 #endif
 #ifdef HAVE_MALLOC_H
+#ifdef __APPLE__
+#include <stdlib.h>
+#else
 #include <malloc.h>
 #endif
+#endif
 #ifdef HAVE_LIMITS_H
 #include <limits.h>
 #endif
diff -ru a/libiberty/pex-unix.c b/libiberty/pex-unix.c
--- a/libiberty/pex-unix.c
+++ b/libiberty/pex-unix.c
@@ -55,7 +55,7 @@
 #ifdef HAVE_SYS_STAT_H
 #include <sys/stat.h>
 #endif
-#ifdef HAVE_PROCESS_H
+#if defined(HAVE_PROCESS_H) && !defined(__APPLE__)
 #include <process.h>
 #endif
 #ifdef HAVE_SPAWN_H
@@ -67,7 +67,7 @@
 #else
 # define VFORK_STRING "vfork"
 #endif
-#ifdef HAVE_VFORK_H
+#if defined(HAVE_VFORK_H) && !defined(__APPLE__)
 #include <vfork.h>
 #endif
 #if defined(VMS) && defined (__LONG_POINTERS)
diff -ru a/libiberty/physmem.c b/libiberty/physmem.c
--- a/libiberty/physmem.c
+++ b/libiberty/physmem.c
@@ -25,20 +25,20 @@
 # include <unistd.h>
 #endif
 
-#if HAVE_SYS_PSTAT_H
+#if HAVE_SYS_PSTAT_H && !defined(__APPLE__)
 # include <sys/pstat.h>
 #endif
 
-#if HAVE_SYS_SYSMP_H
+#if HAVE_SYS_SYSMP_H && !defined(__APPLE__)
 # include <sys/sysmp.h>
 #endif
 
-#if HAVE_SYS_SYSINFO_H && HAVE_MACHINE_HAL_SYSINFO_H
+#if HAVE_SYS_SYSINFO_H && HAVE_MACHINE_HAL_SYSINFO_H && !defined(__APPLE__)
 # include <sys/sysinfo.h>
 # include <machine/hal_sysinfo.h>
 #endif
 
-#if HAVE_SYS_TABLE_H
+#if HAVE_SYS_TABLE_H && !defined(__APPLE__)
 # include <sys/table.h>
 #endif
 
@@ -52,7 +52,7 @@
 # include <sys/sysctl.h>
 #endif
 
-#if HAVE_SYS_SYSTEMCFG_H
+#if HAVE_SYS_SYSTEMCFG_H && !defined(__APPLE__)
 # include <sys/systemcfg.h>
 #endif
 
diff -ru a/libiberty/setproctitle.c b/libiberty/setproctitle.c
--- a/libiberty/setproctitle.c
+++ b/libiberty/setproctitle.c
@@ -20,7 +20,7 @@
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
-#ifdef HAVE_SYS_PRCTL_H
+#if defined(HAVE_SYS_PRCTL_H) && !defined(__APPLE__)
 #include <sys/types.h>
 #include <sys/prctl.h>
 #endif
diff -ru a/Makefile.in b/Makefile.in
--- a/Makefile.in
+++ b/Makefile.in
@@ -14400,7 +14400,7 @@
 	  $$s/$$module_srcdir/configure \
 	  --srcdir=$${topdir}/$$module_srcdir \
 	  $(HOST_CONFIGARGS) --build=${build_alias} --host=${host_alias} \
-	  --target=${target_alias} --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@ \
+	  --target=${target_alias} --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@ \
 	  || exit 1
 @endif mpfr
 
@@ -14436,7 +14436,7 @@
 	  --target=${target_alias} \
 	   \
 	  $(STAGE1_CONFIGURE_FLAGS) \
-	  --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@
+	  --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@
 @endif mpfr-bootstrap
 
 .PHONY: configure-stage2-mpfr maybe-configure-stage2-mpfr
@@ -14470,7 +14470,7 @@
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGE2_CONFIGURE_FLAGS) \
-	  --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@
+	  --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@
 @endif mpfr-bootstrap
 
 .PHONY: configure-stage3-mpfr maybe-configure-stage3-mpfr
@@ -14504,7 +14504,7 @@
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGE3_CONFIGURE_FLAGS) \
-	  --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@
+	  --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@
 @endif mpfr-bootstrap
 
 .PHONY: configure-stage4-mpfr maybe-configure-stage4-mpfr
@@ -14538,7 +14538,7 @@
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGE4_CONFIGURE_FLAGS) \
-	  --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@
+	  --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@
 @endif mpfr-bootstrap
 
 .PHONY: configure-stageprofile-mpfr maybe-configure-stageprofile-mpfr
@@ -14572,7 +14572,7 @@
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEprofile_CONFIGURE_FLAGS) \
-	  --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@
+	  --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@
 @endif mpfr-bootstrap
 
 .PHONY: configure-stagetrain-mpfr maybe-configure-stagetrain-mpfr
@@ -14606,7 +14606,7 @@
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEtrain_CONFIGURE_FLAGS) \
-	  --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@
+	  --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@
 @endif mpfr-bootstrap
 
 .PHONY: configure-stagefeedback-mpfr maybe-configure-stagefeedback-mpfr
@@ -14640,7 +14640,7 @@
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEfeedback_CONFIGURE_FLAGS) \
-	  --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@
+	  --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@
 @endif mpfr-bootstrap
 
 .PHONY: configure-stageautoprofile-mpfr maybe-configure-stageautoprofile-mpfr
@@ -14674,7 +14674,7 @@
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEautoprofile_CONFIGURE_FLAGS) \
-	  --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@
+	  --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@
 @endif mpfr-bootstrap
 
 .PHONY: configure-stageautofeedback-mpfr maybe-configure-stageautofeedback-mpfr
@@ -14708,7 +14708,7 @@
 	  --target=${target_alias} \
 	  --with-build-libsubdir=$(HOST_SUBDIR) \
 	  $(STAGEautofeedback_CONFIGURE_FLAGS) \
-	  --disable-shared @extra_mpfr_configure_flags@ @host_libs_picflag@
+	  --disable-shared @extra_mpfr_configure_flags@ --disable-float128 @host_libs_picflag@
 @endif mpfr-bootstrap
 
 
diff -ru a/mpc/Makefile.in b/mpc/Makefile.in
--- a/mpc/Makefile.in
+++ b/mpc/Makefile.in
@@ -376,7 +376,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = src tests doc tools
+SUBDIRS = src
 EXTRA_HEADERS = src/mpc-log.h
 include_HEADERS = src/mpc.h @MPC_LOG_H@
 EXTRA_DIST = doc/fdl-1.3.texi src/mpc-log.h
diff -ru a/mpfr/Makefile.in b/mpfr/Makefile.in
--- a/mpfr/Makefile.in
+++ b/mpfr/Makefile.in
@@ -406,7 +406,7 @@
 # libtoolize and in case some developer needs to switch back to an
 # old Automake version.
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = doc src tests tune tools/bench
+SUBDIRS = src
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = mpfr.pc
 nobase_dist_doc_DATA = AUTHORS BUGS COPYING COPYING.LESSER NEWS TODO \
